cns/kubecontroller/nodenetworkconfig/conversion_windows.go

```diff
@@ -6,14 +6,15 @@ import (
 
 	"github.com/Azure/azure-container-networking/cns"
 	"github.com/Azure/azure-container-networking/crd/nodenetworkconfig/api/v1alpha"
+	"github.com/pkg/errors"
 )
 
 // createNCRequestFromStaticNCHelper generates a CreateNetworkContainerRequest from a static NetworkContainer.
 // If the NC's DefaultGateway is empty, it will set the 0th IP as the gateway IP and all remaining IPs as
 // secondary IPs. If the gateway is not empty, it will not reserve the 0th IP and add it as a secondary IP.
 //
 //nolint:gocritic //ignore hugeparam
-func createNCRequestFromStaticNCHelper(nc v1alpha.NetworkContainer, primaryIPPrefix netip.Prefix, subnet cns.IPSubnet) *cns.CreateNetworkContainerRequest {
+func createNCRequestFromStaticNCHelper(nc v1alpha.NetworkContainer, primaryIPPrefix netip.Prefix, subnet cns.IPSubnet) (*cns.CreateNetworkContainerRequest, error) {
 	secondaryIPConfigs := map[string]cns.SecondaryIPConfig{}
 
 	// if NC DefaultGateway is empty, set the 0th IP to the gateway and add the rest of the IPs
```

Comment:
Let's put this code of Matt to have the extra check for the subnet type to be Overlay specifically and change the logic according for Vnet Block.

Reply:
so not default gateway config needed if it is empty for vnetscale?